### ISSUE TITLE
Is there a way to pass a sourcemap to ngAnnotate?

### ISSUE THREAD

**Issue by thiswildorchid** [mzgol/grunt-ng-annotate#33]
I am trying to concat, then annotate, then minify and I need to create source-map along the way but not sure I can pass a source-map to ng-annotate. Is this possible?

**Comment by mzgol** [comment]
Currently `grunt-ng-annotate` combines source maps but AFAIK only if the input source map is inline in the generated file. This is mostly handled by [ng-annotate](https://github.com/olov/ng-annotate), though, I'm mostly just constructing proper options so that it works. Maybe open an issue there? I can add support but only if `ng-annotate` handles it correctly.

Note, though, that `grunt-ng-annotate` doesn't support creating a source map for multiple source files (see #26). This shouldn't be an issue if you're concatenating using another task, I guess.

**Comment by stefan--** [comment]
@mzgol do I understand you correctly that the following scenario should work?

1. concat several files and generate an inline source map
2. annotate the concatenated file and create a source map
3. the resulting source map should be a combination of both source maps

**Comment by mzgol** [comment]
@stefan-- Interesting; I have a test for combining source maps that ensures combining with Babel inline source maps works:
https://github.com/mzgol/grunt-ng-annotate/blob/259f56efd9f51a32ac1e54ea50f640bcca0efc29/test/spec/api.js#L116-L138.
Both the fact that `sources` point to the original file and the mappings are checked.

Could you post a simple test case showing the bug?

**Comment by stefan--** [comment]
@mzgol thanks for your quick response. I just had a look at your test case in order to understand whether it maps to my case and it does with the only difference that the sourcemap from the transpilation has only one source and the sourcemap from my concatenation task has several sources. In addition, babel is capable of adding the `sourcesRoot` property whereas `grunt-contrib-concat` does not support it yet. Any further suggestions?

**Comment by mzgol** [comment]
Hmm, OK, it seems a basic example with `grunt-contrib-concat` should reproduce the problem then. I'll see what I can do (once I find time, it's been busy for me recently).

One thing you could do is to try to run [ng-annotate](https://www.npmjs.com/package/ng-annotate) directly on the output of `grunt-contrib-concat` with proper parameters and see what needs to be passed there to make it work. This is to determine whether the bug lies in the Grunt plugin or `ng-annotate` itself.

**Comment by stefan--** [comment]
@mzgol it works with `ng-annotate`

**Comment by mzgol** [comment]
Could you write how you call it (with what parameters)? I'll check later what I do incorrectly here.

For reference, this is where `ng-annotate` is called:
https://github.com/mzgol/grunt-ng-annotate/blob/259f56efd9f51a32ac1e54ea50f640bcca0efc29/tasks/ng-annotate.js#L168

**Comment by stefan--** [comment]
@mzgol just 'solved' the problem. I attempted to save the source map not inline but as a separate file with `sourceMap: '/path/to/file'. This is where it fails. Maybe the documentation should take this into account or the lib should support this case. Thanks for your help!

**Comment by mzgol** [comment]
Ah, so it breaks not with inline source maps but with external ones?

That's possible, I have a test for combining inline source maps but no test for combining external ones... Does that work with pure `ng-annotate`? If so, could you post how a snippet showing how you call `ng-annotate`?

**Comment by stefan--** [comment]
@mzgol exactly. It works with pure `ng-annotate` with the following command `ng-annotate -a -o app.annotated.js --sourcemap app.js`, however, it does only support an inline source map, as far as I know.

**Comment by SuneRadich** [comment]
Any news on this issue?

**Comment by mgol** [comment]
No news. I'll happily merge a PR if someone submits one.
